app/models/doi.rb

```diff
@@ -6,6 +6,14 @@ class Doi
 
   MANAGED_PREFIXES = ['10.26207', '10.18113'].freeze
 
+  # @note If the DOI is valid from Datacite's point of view, it will be valid from our perspective, as well as have one
+  # of our managed prefixes, _and_ have a particular format that the Datacite API can interpret.
+  def self.from_datacite?(doi)
+    formatted_doi = new(doi)
+
+    formatted_doi.valid? && formatted_doi.managed? && doi == "#{formatted_doi.prefix}/#{formatted_doi.suffix}"
+  end
+
```

Comment:
@awead - this is probably just a case of me not understanding the full context, but why is this written as a static method rather than an instance method (sorry; prob wrong terminology for Ruby) such as `valid?`, `managed?` etc?

Also, maybe it makes sense to call this something like `datacite_valid?` or something. The name `from_datacite?` to me implies that this method indicates the DOI originated from Datacite, and not that the DOI is valid from Datacite's perspective.

If I'm understanding correctly, all the DOIs in the system should now have this new format that is valid in Datacite, but that does not necessarily mean that all of the DOIs were created by Datacite.

Again, I may be super wrong about some of these assumptions haha. I don't have a super strong opinion on this, but wanted it to throw it out there for your consideration.

Reply:
No, those are good questions. 

To answer the first question, the `from_datacite?` method is a class method. So we'd call is like:

``` ruby
Doi.from_datacite?('doi')`
```

Within the context of the method, we're in the Doi _class_ scope. To get to instance methods, we create an _instance_ of the Doi class (formatted_doi), and then call instance-level methods on it. If we just called `valid?` it would assume the class scope and look for `Doi.valid?`.

Now to your second question... `from_datacite` is misleading. I was originally thinking the "from" part is because that's what is returned from the API. But, that's isn't very clear. Here's what I propose:

An new instance method:

``` ruby
def to_datacite
  "#{prefix}/#{suffix}"
end
```

And rename the class method to:

``` ruby 
def valid_datacite_doi?(doi)
  formatted_doi = new(doi)

  formatted_doi.valid? && formatted_doi.managed? && doi == formatted_doi.to_datacite
end
```

Reply:
Sounds good to me! I think `valid_datacite_doi?` makes its purpose much clearer.